driver/ocamlcomp.sh.in

```diff
@@ -15,6 +15,6 @@
 #*                                                                        *
 #**************************************************************************
 
-topdir=`dirname $0`
+topdir=$(dirname $0)
```

Comment:
Unless the script explicitly uses bash, `$(` is relevantly less portable than back-ticks (some Solaris users, csh users, etc.)

Reply:
I think it's the opposite, see [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006) and the related resources sections.

Reply:
$() is fully portable to all POSIX shells. No OS has deployed a shell that does not understand it in over 20 years.

Reply:
@zapashcanon - in the contexts in which backticks are being used, they work. Changing it brings none of the benefits (because we don't need to nest them and we don't hit the escaping problems, and so forth) but it does break the scripts for *some* users.

@pmetzger - that's simply not true. Solaris 10's `sh` does not understand this syntax, and it's a tier 2 platform.

Reply:
Solaris 10's sh isn't POSIX conformant?

Reply:
I don't have access to Solaris 10, but according to [this](https://unix.stackexchange.com/a/218073) and other tickets say this is so.

Note that /bin/sh can be non-POSIX but the system still be POSIX-compliant.

Reply:
I'd forgotten about the entire Solaris mess. It was the only real exception to the rule.

Reply:
Sure - it may be the only real exception, but it is *an* exception and that for me is enough not to change it, given that as far as I'm aware, there is no need to beyond linting. (I do add that I of course use `$()` in anything new and only use backticks in error, or interactively as it's one character fewer)

Reply:
@dra27 I don't disagree. If portability to Solaris 10 is needed, then this is needed.